doc/components/leading_digit_anticipate.md

```diff
@@ -0,0 +1,41 @@
+# Leading Digit Anticipation
+
+An anticipator is a circuit that can run parallel to another circuit and predict an output: for example, a leading one anticipator can run alongside an adder and predict the leading '1' position of the final sum based on the inputs.  Here are two such circuits:
+
+## LeadingDigitAnticipate
+
+ROHD-HCL comes with a leading-digit anticipate component which predicts the number of leading ones or zeros before the first digit change of a future sum of two inputs.  This prediction is within 1 of the actual digit change (either the `leadingDigit` or `leadingDigit` + 1 is the actual position). This can be used in parallel with an adder to avoid having to do a leading-digit detect after the addition and save delay.
+
+`LeadingDigitAnticipate` assumes a twos-complement representation of the inputs. It will provide a single set of outputs `leadingDigit`, the position of the first `1` for a positive sum and the first '0' for a negative sum, as well as `validLeadDigit` which indicates that a digit change was predicted.  
+
+Here is a sample usage:
+
+```dart
+final predictor = LeadingDigitAnticipate(a, b);
+
+final sumShift = (a + b) << predictor.leadingDigit;
```

Comment:
this example usage is good, but i think it's missing some context as to why you'd want to shift by `leadingDigit` amount?  i.e. maybe explain that the goal is for there to be no leading 0s?